rest/admin_api.go

```diff
@@ -350,7 +351,10 @@ func (h *handler) handlePostIndexInit() error {
 		}
 	}
 
-	useLegacySyncDocsIndex := h.db.UseLegacySyncDocsIndex() // TODO: CBG-4607: Change when this API is updated to support the new principal indexes
+	useLegacySyncDocsIndex := h.db.UseLegacySyncDocsIndex()
+	if req.SeparatePrincipalIndexes != nil {
+		useLegacySyncDocsIndex = !(*req.SeparatePrincipalIndexes)
```

Comment:
If this doesn't change the value of useLegacySyncDocsIndex (from line 354), do you think we should be returning an error similar to the num_partitions handling on line 326?

Reply:
I thought about this and this was in Ben's PR originally. I do like making an API more restrictive.

The reason _not_ to do this is if a database is running but an index was deleted, this would let you recreate it without taking the database down.

This is a pretty niche case and I can be swayed either way. Since it is pretty much idempotent and pretty quick if all the indexes are recreated, and it doesn't put a lot of load on the system, this didn't seem bad.

Reply:
I'm caught up on this now. I'm certainly open to the idea of removing the error for `num_partitions` and letting the request run through init/readiness checks again.